### PR TITLE
from hail import * -> import hail as hl

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Sorry for the large diff, it's a conceptually small change that required a lot of refactoring!

File: python/hail/__init__.py [hunk 1]
```diff
@@ -32,30 +42,9 @@
...
-           'TSet',
-           'TDict',
-           'TStruct',
-           'TLocus',
-           'TCall',
-           'TInterval',
-           'KinshipMatrix']
+           'agg']
+
+__all__.extend(genetics.__all__)
```
Comment (jigold) [inline python/hail/__init__.py]: nice!

File: python/hail/docs/conf.py [hunk 2]
```diff
@@ -71,15 +71,13 @@
...
-stop()
-
-if os.path.isdir("output/"):
-    shutil.rmtree("output/")
-
-files = ["sample.vds", "sample.qc.vds", "sample.filtered.vds"]
-for f in files:
-    if os.path.isdir(f):
-        shutil.rmtree(f)
+# if os.path.isdir("output/"):
```
Comment (jigold) [inline python/hail/docs/conf.py]: delete?

File: python/hail/expr/expression.py [hunk 3]
```diff
@@ -885,9 +901,7 @@ def product(self):
         :class:`.NumericExpression`
             The product of the collection.
         """
-        return self._method("product",
-                            TInt64() if isinstance(self._type.element_type, TInt32) or
-                                        isinstance(self._type.element_type, TInt64) else TFloat64())
+        return self._method("product", self._type.element_type)
```
Comment (jigold) [inline python/hail/expr/expression.py]: Does the type conversion get handled somewhere else now?

File: python/hail/methods/impex.py [hunk 4]
```diff
@@ -453,13 +453,13 @@ def import_bed(path, reference_genome=None):
     # Add the variant annotation ``va.cnvRegion: Boolean`` indicating inclusion in
     # at least one interval of the three-column BED file `file1.bed`:
 
-    # >>> bed = methods.import_bed('data/file1.bed')
+    # >>> bed = hl.import_bed('data/file1.bed')
```
Comment (jigold) [inline python/hail/methods/impex.py]: Should this be uncommented?

File: python/hail/docs/doctest_setup.py [hunk 5]
```diff
@@ -10,66 +12,28 @@
...
-ds = ds.sample_rows(0.015)
-ds = ds.annotate_rows(anno1 = 5, toKeep1 = True, toKeep2 = False, toKeep3 = True)
-ds = methods.split_multi_hts(ds)
-ds.write('data/example2.vds', overwrite=True)
-
-ds = methods.import_vcf('data/sample.vcf.bgz')
-ds.write('data/example2.multi.generic.vds', overwrite=True)
-
-ds = methods.import_vcf('data/sample.vcf.bgz')
-ds = methods.split_multi_hts(ds)
```
Comment (jigold) [inline python/hail/docs/doctest_setup.py]: did we not need all of these imports?

File: python/hail/expr/expression.py (hunk 3 above)
Comment (tpoterba) [reply python/hail/expr/expression.py]: this was just wrong. product returns an array of the same type.

File: python/hail/methods/impex.py (hunk 4 above)
Comment (tpoterba) [reply python/hail/methods/impex.py]: yes, but don't want to debug if it's wrong as part of this orthogonal PR

File: python/hail/docs/doctest_setup.py (hunk 5 above)
Comment (tpoterba) [reply python/hail/docs/doctest_setup.py]: I moved the burden + lmmreg ones to the places they're used (once each). This sped up doctests by about 3x.

**Comment by tpoterba** [comment]
waiting on the Call branch to merge this.

**Pull request merged**